lang/cs/Org.Apache.REEF.IMRU/OnREEF/Driver/TaskManager.cs

```diff
@@ -225,7 +232,14 @@ internal void RecordCompletedTask(ICompletedTask completedTask)
         {
             if (completedTask.Id.Equals(_masterTaskId))
             {
-                if (GetTaskInfo(completedTask.Id).TaskState.CurrentState.Equals(TaskState.TaskRunning))
+                string message = "";
+                if (completedTask.Message != null)
+                {
+                    message = ByteUtilities.ByteArraysToString(completedTask.Message);
+                    Logger.Log(Level.Info, "UpdateTask message {0}", message);
+                }
+
+                if (GetTaskInfo(completedTask.Id).TaskState.CurrentState.Equals(TaskState.TaskRunning) || message.Equals(TaskManager.UpdateTaskCompleted))
```

Comment:
If the update task is closed by driver, the task state will be TaskWaitingForClose rather than TaskRunning, at the same time, the update task has completed, we will only get TaskManager.UpdateTaskCompleted true. 

For successful case, update task is turning from TaskRunning to TaskCompleted, yes, both conditions are true.